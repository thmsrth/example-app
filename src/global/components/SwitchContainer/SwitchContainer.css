.SwitchContainer__options {
  padding: calc(var(--section-padding) * 2);
}