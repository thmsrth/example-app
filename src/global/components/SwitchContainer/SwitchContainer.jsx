import React from 'react';
import PropTypes from 'prop-types';
import { bemCls } from '../../utils/ClassNameHelpers';

import './SwitchContainer.css';

const SwitchContainer = ({
  className, children, options, selected, handleChange,
}) => (
  <div className={`SwitchContainer ${className}`}>
    <div className={`${bemCls(`SwitchContainer ${className}`, '__options')}`}>
      {options.map(option => (
        <div
          key={option.key}
          className={`${bemCls(`SwitchContainer ${className}`, '__option')} ${
            bemCls(`SwitchContainer ${className}`, '__selected', selected === option.key)}`}
        >
          <label className={`${bemCls(`SwitchContainer ${className}`, '__label')}`}>
            <input
              className={`${bemCls(`SwitchContainer ${className}`, '__input')}`}
              type="radio"
              id={option.key}
              value={option.key}
              onChange={() => handleChange(option.key)}
              checked={selected === option.key}
            />
            {option.label}
          </label>
        </div>
      ))}
    </div>
    <div className={`${bemCls(`SwitchContainer ${className}`, '__children')}`}>
      {children.find(child => child.key === selected)}
    </div>
  </div>
);

export default SwitchContainer;

SwitchContainer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  options: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string,
    title: PropTypes.string,
  })),
  selected: PropTypes.string,
  handleChange: PropTypes.func,
};

SwitchContainer.defaultProps = {
  className: '',
  children: null,
  options: [
    {
      key: '1',
      label: 'Option label',
    },
    {
      key: '2',
      label: 'Another label',
    },
    {
      key: '3',
      label: 'Third label',
    },
  ],
  selected: null,
  handleChange: () => {},
};
