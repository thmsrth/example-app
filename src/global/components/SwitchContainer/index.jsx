import React from 'react';
import PropTypes from 'prop-types';
import SwitchContainer from './SwitchContainer';

class SwitchContainerWrapper extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: props.children ? props.children[0].key : '',
    };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(selected) {
    this.setState({ selected: selected.toString() });
  }

  render() {
    const {
      selected,
    } = this.state;
    return (
      <SwitchContainer
        selected={selected}
        handleChange={this.handleChange}
        {...this.props}
      />
    );
  }
}

export default SwitchContainerWrapper;

SwitchContainerWrapper.propTypes = {
  children: PropTypes.node,
};

SwitchContainerWrapper.defaultProps = {
  children: null,
};
