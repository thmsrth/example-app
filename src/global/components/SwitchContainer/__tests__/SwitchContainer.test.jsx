import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SwitchContainer from '../SwitchContainer';

const props = {
  className: 'MySwitchContainer',
  selected: '1',
  options: [
    { key: '1', label: 'Hi' },
    { key: '2', label: 'Ciao' },
  ],
  onChange: () => {},
};

describe('SwitchContainer', () => {
  it('should render the component correctly with option 1 selected and h1 Hi component rendered', () => {
    const component = (
      <SwitchContainer {...props}>
        <h1 key="1">Hi</h1>
        <h1 key="2">Ciao</h1>
      </SwitchContainer>
    );
    const tree = shallow(component);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly with no option selected', () => {
    const component = (
      <SwitchContainer {...props} selected="">
        <h1 key="1">Hi</h1>
        <h1 key="2">Ciao</h1>
      </SwitchContainer>
    );
    const tree = shallow(component);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
