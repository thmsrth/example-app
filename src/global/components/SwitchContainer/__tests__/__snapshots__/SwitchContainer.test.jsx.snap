// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchContainer should render the component correctly with no option selected 1`] = `
<div
  className="SwitchContainer MySwitchContainer"
>
  <div
    className="SwitchContainer__options MySwitchContainer__options"
  >
    <div
      className="SwitchContainer__option MySwitchContainer__option "
      key="1"
    >
      <label
        className="SwitchContainer__label MySwitchContainer__label"
      >
        <input
          checked={false}
          className="SwitchContainer__input MySwitchContainer__input"
          id="1"
          onChange={[Function]}
          type="radio"
          value="1"
        />
        Hi
      </label>
    </div>
    <div
      className="SwitchContainer__option MySwitchContainer__option "
      key="2"
    >
      <label
        className="SwitchContainer__label MySwitchContainer__label"
      >
        <input
          checked={false}
          className="SwitchContainer__input MySwitchContainer__input"
          id="2"
          onChange={[Function]}
          type="radio"
          value="2"
        />
        Ciao
      </label>
    </div>
  </div>
  <div
    className="SwitchContainer__children MySwitchContainer__children"
  />
</div>
`;

exports[`SwitchContainer should render the component correctly with option 1 selected and h1 Hi component rendered 1`] = `
<div
  className="SwitchContainer MySwitchContainer"
>
  <div
    className="SwitchContainer__options MySwitchContainer__options"
  >
    <div
      className="SwitchContainer__option MySwitchContainer__option SwitchContainer__selected MySwitchContainer__selected"
      key="1"
    >
      <label
        className="SwitchContainer__label MySwitchContainer__label"
      >
        <input
          checked={true}
          className="SwitchContainer__input MySwitchContainer__input"
          id="1"
          onChange={[Function]}
          type="radio"
          value="1"
        />
        Hi
      </label>
    </div>
    <div
      className="SwitchContainer__option MySwitchContainer__option "
      key="2"
    >
      <label
        className="SwitchContainer__label MySwitchContainer__label"
      >
        <input
          checked={false}
          className="SwitchContainer__input MySwitchContainer__input"
          id="2"
          onChange={[Function]}
          type="radio"
          value="2"
        />
        Ciao
      </label>
    </div>
  </div>
  <div
    className="SwitchContainer__children MySwitchContainer__children"
  >
    <h1
      key="1"
    >
      Hi
    </h1>
  </div>
</div>
`;
